### PR TITLE
added jsonschema validation to ci

### PR DESCRIPTION
**Pull request by krishnaprasanthg**
The Pull Request resolves Issue #584

Description:
- defined schema for the vocabulary file.
- Added the jsonschema validation to the travis-ci pipeline

### CODE REVIEW

**Comment by krishnaprasanthg** [comment]
with json-schema turned on:

```shell
Checking file A/Abysmal.json...
Checking file A/Adjoin.json...
🔥 Error: ['parts-of-speech'] -> 'verb' is not one of ['Noun', 'Pronoun', 'Verb', 'Adverb', 'Adjective', 'Preposition', 'Conjunction', 'Interjunction']
Checking file A/Ancillary.json...
🔥 Error: ['parts-of-speech'] -> 'adjective' is not one of ['Noun', 'Pronoun', 'Verb', 'Adverb', 'Adjective', 'Preposition', 'Conjunction', 'Interjunction']
Checking file B/Bastardization.json...
Checking file C/Captain.json...
Checking file D/Derive.json...
🔥 Error: ['parts-of-speech'] -> 'verb' is not one of ['Noun', 'Pronoun', 'Verb', 'Adverb', 'Adjective', 'Preposition', 'Conjunction', 'Interjunction']
Checking file D/Drown.json...
Checking file E/Envy.json...
Checking file G/Gator.json...
Checking file H/Harangue.json...
Checking file H/Hoodwink.json...
Checking file H/Huzzahs.json...
Checking file L/Lazy.json...
Checking file M/Melancholy_adjective.json...
Checking file M/Melancholy_noun.json...
Checking file M/Mellifluous.json...
Checking file M/Mullock.json...
Checking file N/Near.json...
🔥 Error: ['parts-of-speech'] -> 'adverb' is not one of ['Noun', 'Pronoun', 'Verb', 'Adverb', 'Adjective', 'Preposition', 'Conjunction', 'Interjunction']
Checking file N/Neptune.json...
Checking file P/Pollyannaish.json...
Checking file S/Sardonic.json...
Checking file T/Tale.json...
🔥 Error: ['word'] -> 'Tale ' does not match '^[A-Z][a-z]+$'
Checking file T/Tip.json...
Checking file V/Velocity.json...
🔥 Error: ['word'] -> 'velocity' does not match '^[A-Z][a-z]+$'
Checking file X/Xenon.json...
Checking file Z/Zenith.json...
🔥 Error: ['word'] -> 'Zenith0' does not match '^[A-Z][a-z]+$'
```

**Comment by Showndarya** [comment]
Hey @krishnaprasanthg 

There is a merge conflict in the PR. Can you resolve it?

Thanks for contributing!

**Comment by jukefr** [comment]
I'll handle it

**Comment by jukefr** [comment]
We are going to need a `requirements.txt` file too and to setup caching for the module for faster boots

File: .travis.py [hunk 1]
```diff
@@ -1,8 +1,35 @@
 import json
+import operator
 import os
 import re
 import subprocess
 
+from jsonschema import Draft4Validator
```
Comment (jukefr) [inline .travis.py]: need to create `requirements.txt` and add the package to it

**Comment by jukefr** [comment]
Also you need to edit the `.travis.yml` file and add caching support for pip

https://docs.travis-ci.com/user/caching/#pip-cache

File: .travis.py (hunk 1 above)
Comment (krishnaprasanthg) [reply .travis.py]: thanks for reviewing. Corrected now.

**Pull request merged**